src/Microsoft.Azure.WebJobs.Host/Configuration/IWebJobsExtensionConfiguration.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Host.Configuration
+{
+    public interface IWebJobsExtensionConfiguration<T> where T : IExtensionConfigProvider
```

Comment:
The configuration.Configuration naming pattern this introduces does look a little strange.

Reply:
Didn't like that, but the type name options aren't great considering some of the other types in the SDK. I've renamed the property to `ConfigurationSection` . We can move forward with a type rename later if needed.